From 8126583357773fc9a06f53291043a69566d4a097 Mon Sep 17 00:00:00 2001
From: KOLANICH <KOLANICH@users.noreply.github.com>
Date: Mon, 10 Feb 2020 00:00:00 +0000
Subject: [PATCH] Fixed buffer overrun via integer underflow.

---
 strutil.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/strutil.cc b/strutil.cc
index a585963..5c4f96c 100644
--- a/strutil.cc
+++ b/strutil.cc
@@ -475,8 +475,9 @@ StringPiece TrimLeadingCurdir(StringPiece s) {
 }
 
 void FormatForCommandSubstitution(string* s) {
-  while ((*s)[s->size() - 1] == '\n')
+  while (s->size() >0 && (*s)[s->size() - 1] == '\n'){
     s->pop_back();
+  }
   for (size_t i = 0; i < s->size(); i++) {
     if ((*s)[i] == '\n')
       (*s)[i] = ' ';
-- 
2.20.1

