From 38c8f2238e26caf95cdfdeb2cc031e51cbbc8325 Mon Sep 17 00:00:00 2001
From: KOLANICH <KOLANICH@users.noreply.github.com>
Date: Mon, 13 Jan 2020 00:00:00 +0000
Subject: [PATCH 3/4] Performance and hardening flags

---
 Makefile.ckati | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.ckati b/Makefile.ckati
index 795aa05..95a7766 100644
--- a/Makefile.ckati
+++ b/Makefile.ckati
@@ -71,11 +71,12 @@ KATI_CXX_TEST_EXES := $(patsubst $(KATI_INTERMEDIATES_PATH)/%.o,$(KATI_BIN_PATH)
 	$(KATI_CXX_TEST_OBJS))
 
 KATI_CXXFLAGS := -g -W -Wall -MMD -MP
+KATI_CXXFLAGS += -D_FORTIFY_SOURCE=2 -D_GLIBCXX_ASSERTIONS -O3 -Wall -Wextra -Wconversion -Wformat -Wformat-security -Werror=format-security -fno-strict-aliasing -fno-common -fstack-check -fcf-protection=full -fsanitize=cfi-cast-strict -mretpoline -mspeculative-load-hardening -fsanitize=undefined -ftrapv -fstack-protector -fstack-protector-strong -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-flto -Wl,-z,relro -Wl,-z,now -Wl,-z,ibtplt -Wl,-z,ibt -Wl,-z,shstk
 KATI_CXXFLAGS += -O -DNOLOG
 #KATI_CXXFLAGS += -pg
 
 ifeq ($(shell uname),Linux)
-KATI_LIBS := -lrt -lpthread
+KATI_LIBS := -lrt -lpthread -L/usr/lib/llvm-11/lib/clang/11.0.0/lib/linux/ -l:libclang_rt.ubsan_standalone-x86_64.so
 endif
 
 # Rule to build ckati into KATI_BIN_PATH
-- 
2.20.1

